### PR TITLE
Replace needless std::map with std::vector, and std::move in capture

### PR DESCRIPTION
**Pull request by greg7mdp**
small optimizations.

### CODE REVIEW

File: plugins/http_plugin/include/eosio/http_plugin/macros.hpp [hunk 1]
```diff
@@ -15,7 +15,7 @@ struct async_result_visitor : public fc::visitor<fc::variant> {
          auto params = parse_params<api_namespace::call_name ## _params, params_type>(body);\
          FC_CHECK_DEADLINE(deadline);\
          api_handle.call_name( std::move(params), \
-            [cb, body](const std::variant<fc::exception_ptr, call_result>& result){\
+               [cb=std::move(cb), body=std::move(body)](const std::variant<fc::exception_ptr, call_result>& result){ \
```
Comment (heifner) [inline plugins/http_plugin/include/eosio/http_plugin/macros.hpp:18]: `cb` and `body` are used in the catch.

File: plugins/producer_api_plugin/producer_api_plugin.cpp [hunk 2]
```diff
@@ -43,7 +43,7 @@ struct async_result_visitor : public fc::visitor<fc::variant> {
 {std::string("/v1/" #api_name "/" #call_name), \
    [&api_handle](string&&, string&& body, url_response_callback&& cb) mutable { \
       if (body.empty()) body = "{}"; \
-      auto next = [cb, body](const std::variant<fc::exception_ptr, call_result>& result){\
+      auto next = [cb=std::move(cb), body=std::move(body)](const std::variant<fc::exception_ptr, call_result>& result){ \
```
Comment (heifner) [inline plugins/producer_api_plugin/producer_api_plugin.cpp:46]: `body` is used by `INVOKE`.

**Review by heifner — CHANGES_REQUESTED** [review]

**Review by linh2931 — APPROVED** [review]

File: plugins/http_plugin/include/eosio/http_plugin/macros.hpp (hunk 1 above)
Comment (greg7mdp) [reply plugins/http_plugin/include/eosio/http_plugin/macros.hpp:18]: Hum, my understanding is that the lambda we construct (into which I mode `cb` and `body`) is passed to the API as `next`, and if the API fails for any reason it invokes the `next()` function and doesn't rethrow. That is why we check for exceptions inside the lambda.

So either we construct the lambda, and we pass it to the API, or there was an exception which happened *before* the API was called in which case the bottom catc clause is used.

File: plugins/producer_api_plugin/producer_api_plugin.cpp (hunk 2 above)
Comment (greg7mdp) [reply plugins/producer_api_plugin/producer_api_plugin.cpp:46]: same here.

**Review by heifner — APPROVED** [review]

**Pull request merged**